cpmonitor/management/commands/cleanupimages.py

```diff
@@ -0,0 +1,212 @@
+import os
+import re
+
+from django.apps import apps
+from django.core.management.base import BaseCommand, CommandError
+from django.core.exceptions import ValidationError
+from django.core.validators import URLValidator
+from django.db import models as DjangoModels
+
+APP_NAME = __package__.split(".")[0]
+BASE_PATH = os.environ.get("BASE_PATH")
+IMAGES_PATH = os.environ.get("IMAGES_PATH")
+
+
+class Command(BaseCommand):
+    help = "Cleans up orphan images that are no longer used but are still linked in Markdown or file fields"
+
+    def add_arguments(self, parser):
```

Comment:
When I run ` python ./manage.py cleanupimages`
I get the error:
```CommandError: Error occurred by cleaning up images: expected str, bytes or os.PathLike object, not NoneType```
Do I need another argument? Or define the settings?

Reply:
You need the BASE_PATH and IMAGES_PATH which are defined in .env. I run the command in Intellij using the .env in the run configuration.

Reply:
Added a check in the command to have a more readable error message

Reply:
The .env files are only supposed to be used by docker compose.

Inside the python app, please use `settings.MEDIA_ROOT` instead (which is used e.g. [where the uploaded images are downscaled](https://github.com/GermanZero-de/klimaschutzmonitor/blob/c2d2b63dabee13bee360e6121b2933ed8ce839ec/cpmonitor/views.py#L497)). That will always be defined without any additional external config, and will point to the right location both locally, in a local docker container and on the server too.

Reply:
This is my first Python/Django project and maybe here it is best practice, but in my opinion it would make more sense to use ENV vars to configure things like this because:

- you could configure different values per stage (local, testing, production)
- the way using ENV vars locally and on testing/production is more consistent 
- the configured values are not hard coded in the code but configurable via [Github environments](https://github.com/GermanZero-de/klimaschutzmonitor/settings/environments), so we don't need a PR when we want to change a value

Reply:
I get your point but I guess as you should in principle be able to use such a management command for any django project, it makes more sense to use the settings rather than our specific setup.
The management command lets you specify which settings to use so that you can have different ones when running locally or on the server.

Reply:
@patrick-local-zero We ready get the advantage you mention in your first bullet point from using the different config files in the repo. Also the situation is a bit more complex than just local vs server, there's also rhe container vs no container difference.
I honestly don't really care which way we do this, but it's working this way now and that took us months. If you have the time to unify it I won't stop you, but I definitely don't.

Reply:
Regarding your second point, I don't think there's a difference right now how env vars are used locally and on the server..? We use a config module in all cases, plus a docker compose .env whenever we run containers. That's the same ln the server and locally.

Your third point is true, but I think needing a PR to change something in the server setup is actually good because it means the history of changes to the setup will be recorded and we cann roll back anytime. That's why I spent so much time trying to get as much as I could of our server setup under version control, and deploy it all every time.

Reply:
The whole deployment was certainly not easy. I didn't mean to question your work over the last few months, sorry about that 🙂  I was just thinking out loud. Now Django settings are used instead of ENV variables.